smartsim/entity/ensemble.py

```diff
@@ -523,15 +523,46 @@ def add_function(
 
     @staticmethod
     def _extend_entity_db_models(model: Model, db_models: t.List[DBModel]) -> None:
-        entity_db_models = [db_model.name for db_model in model.db_models]
+        """
+        Ensures that the Machine Learning model names being added to the Ensemble
+        are unique.
 
+        This static method checks if the provided ML model names already exist in
+        the Ensemble. An SSUnsupportedError is raised if any duplicate names are
+        found. Otherwise, it appends the given list of DBModels to the Ensemble.
+
+        :param model: SmartSim Model object.
+        :type model: Model
+        :param db_models: List of DBModels to append to the Ensemble.
+        :type db_models: t.List[DBModel]
+        """
+        entity_db_models = [db_model.name for db_model in model.db_models]
```

Comment:
nit: can we make this a set so that we can look up if a name is present with the `in` stmt in constant, rather than linear time?
```suggestion
        entity_db_models = {db_model.name for db_model in model.db_models}
```
I don't ever expect this list to be too long, so it _shouldn't reeeaalllyy_ matter, but also doesn't hurt ¯\\\_(ツ)\_/¯

Reply:
I think this will build the whole set no matter what and the set is better but still not a huge improvement. Given that, could we restate more simply:

```py
dupes = next(x.name for x in model.db_models if x.name == input_name, None)
if dupes:
   raise SSUnsupported
```